modules/http/src/main/java/org/mule/module/http/internal/request/grizzly/GrizzlyHttpClient.java

```diff
@@ -515,4 +480,109 @@ public void stop()
     {
         asyncHttpClient.close();
     }
+
+    private class BodyDeferringWorkManagerSource implements AsyncHandler<Response>, WorkManagerSource
```

Comment:
Unsure why "WorkManagerSource" is in the class name, this is really secondary to the purpose of this class.  Instead isn't this a BodyDefferingAsyncHAndler?  (or was name bad before too)?